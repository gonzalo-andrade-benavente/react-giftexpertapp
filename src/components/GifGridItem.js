import PropTypes from 'prop-types';

const GiftGridItem = ({ id, title, img }) => {
    //const GiftGridItem = ( props ) => {

    return (

        <div className="card animate__animated animate__bounce">
            <img src={img} alt={title} />
            <p>{title}</p>
        </div>

    )

}

GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired
}

export default GiftGridItem;