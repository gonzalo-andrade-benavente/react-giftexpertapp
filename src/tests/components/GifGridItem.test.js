import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe( 'Test GifGridItem.test.js', () => {

    const title = 'Soy un t√≠tulo';
    const url = 'https://rutadeprueba/images.js';
    const wrapper = shallow( <GifGridItem title={ title } img={ url }/> );

    test( 'Should be show the Component Correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test( 'Should be a text in markup <p></p>', () => {
        const p = wrapper.find( 'p' );
        expect( p.text().trim() ).toBe( title );
    });

    test( 'Should be a image with url an alt property', () => {
        const img = wrapper.find( 'img' );
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });

    test( 'Should be a effect from css animate__bounce', () => {
        const div = wrapper.find('div');
        expect( div.prop('className').includes('animate__bounce') ).toBe( true );

    });

});