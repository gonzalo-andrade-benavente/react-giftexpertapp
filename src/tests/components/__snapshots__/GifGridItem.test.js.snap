// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGridItem.test.js Should be show the Component Correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Soy un título"
    src="https://rutadeprueba/images.js"
  />
  <p>
    Soy un título
  </p>
</div>
`;
